servicetalk-encoding-netty/src/main/java/io/servicetalk/encoding/netty/NettyCompressionStreamingSerializer.java

```diff
@@ -110,6 +126,34 @@ public void onComplete() {
 
                 subscriber.onComplete();
             }
+
+            private void aggregateNext(Buffer next) {
+                assert aggregate != null;
+                if (!aggregate.isReadOnly() && aggregate.maxWritableBytes() >= next.readableBytes()) {
+                    aggregate.writeBytes(next);
+                } else {
+                    Buffer newAgg = allocator.newBuffer(aggregate.readableBytes() + next.readableBytes(),
```

Comment:
This can overflow

Reply:
yes. we need to feed Netty a `ByteBuf` and since Netty doesn't queue data during the decode (which is why this PR is required) the most pending data we can feed is a single `ByteBuf`. So either we use `CompositeByteBuf` for aggregation or a single buffer with max of 2gb single chunks. Since this is streaming decoder the assumption is chunks can be smaller sizes to avoid overflow.

Reply:
agreed, it's ok to use a single buffer with max 2gb capacity. consider handling this use-case anyway to throw a more informative exception (`BufferEncodingException`?) instead of `IllegalArgumentException` that will come from allocator.